ibazel/command/notify_command.go

```diff
@@ -88,6 +88,14 @@ func (c *notifyCommand) Start(logFile *os.File) (*bytes.Buffer, error) {
 	return outputBuffer, nil
 }
 
+func (c *notifyCommand) RestartProcess() {
```

Comment:
I am wondering how does this function restart the process? Isn't it just writing a stdin?

Reply:
You're right. There are two types of commands : DefaultCommand and NotifyCommand. For Default, RestartProcess() actually just calls Terminate(), but Notify doesn't actually want to kill its process, it simply sends a message. Because both of them are extensions of Command, they have to share the same functions, so I named this one Restart even though it doesn't really restart for Notify.

Reply:
So calling `DefaultCommand.terminate()` is how we actually stop serving old code?

Reply:
Yes. But we only want to stop serving old code on DefaultCommands, not on NotifyCommands. Previously, the old NotifyofChange() function was doing the functionality of the new RestartProcess() and NotifyofChange() functions, so I just split the functionality so the killing would happen sooner, and the second half remained in the same place

Reply:
Yeah, I think the names here should be changed to something more generic like `BeforeRebuild` and `AfterRebuild`